ddl/partition.go

```diff
@@ -978,6 +978,44 @@ func onTruncateTablePartition(d *ddlCtx, t *meta.Meta, job *model.Job) (int64, e
 		}
 	}
 
+	var rules []*placement.RuleOp
+
+	rules, err = infosync.GetPlacementRules(nil)
```

Comment:
They may be millions of rules in future.

Reply:
Getting by group does not improve anything. Because PD does not store rules by group. Anyway it is a full iteration, even with the new api that I am going to use in the next PR.

Reply:
My point is, the result set returned is too large.

Reply:
Yeah. But now we have only one group `TIDB_DDL`. It is not trivial to support it.
It invokes many modifications. That is also why I am pushing it to the next PR.

Reply:
Why not fetch placement rules of only one partition each time? Is it relevant to modification of PD?

Reply:
You can. (PD has merged that PR). But the problem is that we have been using one group from the beginning. And to get only one partition's rules, you need to refactor all previous code.

Reply:
Understood.